source/fundamentals/data-formats/serialization.txt

```diff
@@ -0,0 +1,190 @@
+discriminators.
+
+.. note::
+
+    To use the ``Codec`` interface to specify custom encoding and decoding of 
+    Kotlin objects to BSON data, see the :ref:`Codecs <fundamentals-codecs>` 
+    documentation.
+
+Although you can use the Kotlin driver with the Kotlin Serialization's ``Json`` 
+library, the ``Json`` serializer does *not* directly support BSON value types such
+as ``ObjectId``. You must provide a custom serializer that can handle the 
+conversion between BSON and JSON. 
+
+Supported Types
+~~~~~~~~~~~~~~~
+
+The Kotlin driver supports all Kotlin types that are supported by the Kotlin 
+Serialization library and the following BSON types:
+
+- ObjectId
+- BsonNull
+- BsonArray
+- BsonBinary
+- BsonBoolean
+- BsonDateTime
+- BsonDbPointer
+- BsonDocument
+- BsonDouble
+- BsonInt32
+- BsonInt64
+- BsonDecimal128
+- BsonMaxKey
+- BsonMinKey
+- BsonJavaScript
+- BsonJavaScriptWithScope
+- BsonObjectId
+- BsonRegularExpression
+- BsonString
+- BsonSymbol
+- BsonTimestamp
+- BsonUndefined
+- BsonDocument
+- RawBsonDocument
+- RawBsonArray
+
+Any additional types can be supported using custom serializers. See the 
+`official Kotlin Documentation <https://kotlinlang.org/docs/serialization.html>`__ 
+for more information.
+
+Add Kotlin Serialization to Your Project
+----------------------------------------
+
+The Kotlin driver serialization support depends on the the official Kotlin 
+Serialization library. You must add `Kotlin Serialization <https://github.com/Kotlin/kotlinx.serialization>`__ 
+to your project:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Gradle
+
+      If you are using `Gradle <https://gradle.org/>`__, add the following to your
+      ``build.gradle.kts`` dependencies list:
+
+      .. code-block:: kotlin
+        :caption: build.gradle.kts
+
+        implementation("org.jetbrains.kotlinx:kotlinx-serialization-core:{+serializationVersion+}")
+
+   .. tab::
+      :tabid: Maven
+
+      If you are using `Maven <https://maven.apache.org/>`__, add the following to
+      your ``pom.xml`` dependencies list:
+
+      .. code-block:: kotlin
+        :caption: pom.xml
+   
+        <dependency>
+            <groupId>org.jetbrains.kotlinx</groupId>
+            <artifactId>kotlinx-serialization-core</artifactId>
+            <version>{+serializationVersion+}</version>
+        </dependency>
+
+Annotate Data Classes
+---------------------
+
+To declare a class as serializable, annotate your Kotlin data classes with the 
+``@Serializable`` annotation that comes from the Kotlin Serialization framework.
+
+Once declared as serializable, use your data classes in your code as normal. 
+The Kotlin driver and the Kotlin Serialization framework will handle the 
+BSON serialization and deserialization.
+
+.. note:: 
+
+    You cannot use :ref:`annotations <fundamentals-data-class-annotations>` 
+    from the ``org.bson.codecs.pojo.annotations`` package on ``@Serializable`` data classes.
```

Comment:
S: move this note after the code snippet

Reply:
moved